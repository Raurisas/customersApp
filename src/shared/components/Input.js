import React from 'react';

const Input = props => {
  const { name, type, value, handleChange, placeholder, required } = props;
  return (
    <input
      className="default-input"
      id={name}
      name={name}
      type={type}
      value={value}
      onChange={handleChange}
      placeholder={placeholder}
      required={required}
    />
  );
};

export default Input;
