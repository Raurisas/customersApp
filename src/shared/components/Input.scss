.default-input {
  border: 1px solid $border-color;
  border-radius: 0;
  width: 100%;
  margin: 0 0;
  margin-top: 10px;
  height: 40px;
  padding: 10px;
  outline: none;
}