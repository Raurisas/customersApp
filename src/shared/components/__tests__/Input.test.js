import React from 'react';
import { shallow } from 'enzyme';
import Input from '../Input';

it('should match snapshot', () => {
  const name = 'name';
  const type = 'type';
  const value = 'value';
  const placeholder = 'placeholder';
  const required = true;

  const wrapper = shallow(
    <Input
      name={name}
      type={type}
      value={value}
      placeholder={placeholder}
      required={required}
      handleChange={() => {}}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
