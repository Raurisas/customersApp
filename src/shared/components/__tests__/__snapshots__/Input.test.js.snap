// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<input
  className="default-input"
  id="name"
  name="name"
  onChange={[Function]}
  placeholder="placeholder"
  required={true}
  type="type"
  value="value"
/>
`;
